src/Tests/Component/Signing/AuthnRequestSigningTest.php

```diff
@@ -0,0 +1,216 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\SamlBundle\Tests\Component\Signing;
+
+use PHPUnit_Framework_TestCase as TestCase;
+use Psr\Log\NullLogger;
+use SAML2_AuthnRequest;
+use SAML2_Certificate_KeyLoader;
+use SAML2_Certificate_X509;
+use SAML2_DOMDocumentFactory;
+use Surfnet\SamlBundle\SAML2\AuthnRequest;
+use Surfnet\SamlBundle\Signing\SignatureVerifier;
+use XMLSecurityKey;
+
+class AuthnRequestSigningTest extends TestCase
+{
+    /**
+     * @var string
+     */
+    private $authRequestNoSubject = <<<AUTHNREQUEST_NO_SUBJECT
+<samlp:AuthnRequest
+    xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol"
+    xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion"
+    AssertionConsumerServiceIndex="1"
+    Destination="https://tiqr.stepup.org/idp/profile/saml2/Redirect/SSO"
+    ID="_2b0226190ca1c22de6f66e85f5c95158"
+    IssueInstant="2014-09-22T13:42:00Z"
+    Version="2.0">
+  <saml:Issuer>https://gateway.stepup.org/saml20/sp/metadata</saml:Issuer>
+</samlp:AuthnRequest>
+AUTHNREQUEST_NO_SUBJECT;
+
+    /**
+     * @var string|null
+     */
+    private $publicKey = null;
+
+    /**
+     * @test
+     * @group Signing
+     */
+    public function a_signature_is_verified_if_the_data_to_sign_corresponds_with_how_it_is_included_in_the_http_query_regardless_of_how_it_is_encoded(
+    )
+    {
+        $authnRequestWithDefaultEncoding = $this->createSignedAuthnRequestPreparingDataToSignWith(
+            [$this, 'encodeDataToSignWithPhpsHttpBuildQuery']
+        );
+        $authnRequestWithCustomEncoding  = $this->createSignedAuthnRequestPreparingDataToSignWith(
+            [$this, 'encodeDataToSignWithCustomEncoder']
+        );
+
+        $certificate = SAML2_Certificate_X509::createFromCertificateData($this->getPublicKey());
+
+        $signatureVerifier    = new SignatureVerifier(new SAML2_Certificate_KeyLoader, new NullLogger);
+        $signatureWithDefaultEncodingIsVerified = $signatureVerifier->isSignedWith(
+            $authnRequestWithDefaultEncoding,
+            $certificate
+        );
+        $signatureWithCustomEncodingIsVerified  = $signatureVerifier->isSignedWith(
+            $authnRequestWithCustomEncoding,
+            $certificate
+        );
+
+        $this->assertTrue(
+            $signatureWithDefaultEncodingIsVerified,
+            'The signature of the authn request with PHP\'s default http query encoding '
+            . 'should correspond with its data to sign, but it does not'
+        );
+        $this->assertTrue($signatureWithCustomEncodingIsVerified,
+            'The signature of the authn request with a custom http query encoding '
+            . 'should correspond with its data to sign, but it does not'
```

Comment:
These messages can use some love. What about:

 * `The signature of an AuthnRequest signed using data-to-sign encoded according to RFC1738 should be verifiable, but it isn't`
 * `The signature of an AuthnRequest signed using data-to-sign encoded using a custom encoding should be verifiable, but it isn't`